tests/run-make/hotpatch-unaffected/rmake.rs

```diff
@@ -0,0 +1,51 @@
+// Check if hotpatch leaves the functions that are already hotpatchable untouched
+
+//@ revisions: x32 x64 aarch64
+//@[x32] only-x86
+//@[x64] only-x86_64
+//@[aarch64] only-aarch64
```

Comment:
Problem: unfortunately run-make tests don't support revisions because it's unclear how they would work with arbitrary Rust programs. You'll need to use conditional compilation e.g.

```rs
fn base_rustc(target: &str) -> Rustc {
    // ...
    rustc.target(target);
    // ...
}

// ...

#[cfg(target_arch = "x86")]
{
    // codegen
    // dump lib
    // do asserts
}
```